modular_zubbers/code/modules/storytellers/core/storyteller_balancer.dm

```diff
@@ -0,0 +1,407 @@
+
+	if(inputs.antag_count() <= 0)
+		return 0
+
+	// Pull metrics from vault (0-3 unless noted)
+	var/activity_level = vault[STORY_VAULT_ANTAGONIST_ACTIVITY] || 0
+	var/kills = vault[STORY_VAULT_ANTAG_KILLS] || 0
+	var/objectives = vault[STORY_VAULT_ANTAG_OBJECTIVES_COMPLETED] || 0
+	var/disruption = vault[STORY_VAULT_ANTAG_DISRUPTION] || 0
+	var/influence = vault[STORY_VAULT_ANTAG_INFLUENCE] || 0
+	var/dead_ratio = vault[STORY_VAULT_ANTAG_DEAD_RATIO] || 0
+	var/presence = vault[STORY_VAULT_ANTAGONIST_PRESENCE] || 0
+	var/inactive_ratio = vault[STORY_VAULT_ANTAG_INACTIVE_RATIO] || 0
+	var/intensity = vault[STORY_VAULT_ANTAG_INTENSITY] || 0
+	var/teamwork = vault[STORY_VAULT_ANTAG_TEAMWORK] || 0
+	var/stealth = vault[STORY_VAULT_ANTAG_STEALTH] || 0
+	var/escalation = vault[STORY_VAULT_THREAT_ESCALATION] || 0
+
+	var/weight_activity = 0.15
+	var/weight_kills = 0.10
+	var/weight_objectives = 0.15
+	var/weight_disruption = 0.10
+	var/weight_influence = 0.10
+	var/weight_intensity = 0.10
+	var/weight_teamwork = 0.05
+	var/weight_escalation = 0.10
+	var/weight_stealth = 0.05
+
+	// Normalize to 0-1 (divide by 3 for 0-3 scales; inactive already 0-1)
+	var/norm_activity = activity_level / 3
+	var/norm_kills = kills / 3
+	var/norm_objectives = objectives / 3
+	var/norm_disruption = disruption / 3
+	var/norm_influence = influence / 3
+	var/norm_intensity = intensity / 3
+	var/norm_teamwork = teamwork / 3
+	var/norm_escalation = escalation / 3
+	var/norm_stealth = stealth / 3
+	var/norm_presence = presence / 3
+	var/norm_dead = 1 - (dead_ratio / 3)  // Inverted: high deaths = low survival
+	var/norm_inactive = 1 - inactive_ratio
+
+	// Core strength: weighted sum of key metrics (0-1)
+	var/core_strength = norm_activity * weight_activity + \
+		norm_kills * weight_kills + \
+		norm_objectives * weight_objectives + \
+		norm_disruption * weight_disruption + \
+		norm_influence * weight_influence + \
+		norm_intensity * weight_intensity + \
+		norm_teamwork * weight_teamwork + \
+		norm_escalation * weight_escalation + \
+		norm_stealth * weight_stealth
+
+	// Modifiers (additive multipliers, properly scaled)
+	var/presence_mod = norm_presence * 0.15  // Reduced weight for balance
+	var/survival_mod = norm_dead * 0.20      // Reduced weight for balance
+	var/activity_mod = norm_inactive * 0.25  // Reduced weight for balance
+
+	// Combine core strength with modifiers (weighted sum, no arbitrary centering)
+	var/overall_strength_norm = clamp(
+		core_strength * 0.6 + \
+		(core_strength * presence_mod) * 0.1 + \
+		(core_strength * survival_mod) * 0.15 + \
+		(core_strength * activity_mod) * 0.15,
+		0, 1
+	)
+
+	// Relative scale: antag_count vs. player_count (ideal 5-20% antags; scale 0.5-2.0)
+	var/relative_scale = clamp(inputs.antag_count() / max(inputs.player_count() * 0.1, 1), 0.5, 2.0)
+
+	// Final raw strength: normalize * scale * 100
+	var/final_strength = clamp(overall_strength_norm * relative_scale * 100, 0, 100)
+
+	return final_strength
```

Comment:
at some later point it may really be worth generalizing these values into a way that includes modifers